src/internal_modules/roc_rtp/validator.cpp

```diff
@@ -21,27 +22,30 @@ Validator::Validator(packet::IReader& reader,
     , sample_spec_(sample_spec) {
 }
 
-packet::PacketPtr Validator::read() {
-    packet::PacketPtr next_packet = reader_.read();
-    if (!next_packet) {
-        return NULL;
+int Validator::read(packet::PacketPtr& pp) {
+    packet::PacketPtr next_packet;
+    const int code = reader_.read(next_packet);
+    if (code < 0) {
+        return code;
     }
 
     if (!next_packet->rtp()) {
         roc_log(LogDebug, "rtp validator: unexpected non-rtp packet");
-        return NULL;
+        return error::ErrUnknown;
     }
 
     if (has_prev_packet_ && !validate_(prev_packet_rtp_, *next_packet->rtp())) {
-        return NULL;
+        return error::ErrUnknown;
     }
```

Comment:
Validator should not terminate session (return error) when it sees invalid packet. Instead, it should just drop this specific packet and continue.

If in result validator drops all packets or too much packets, watchdog will terminate session. However if validator drops only some packets, session will work.